src/frontend/apps/impress/src/features/left-panel/components/LeftPanelHeaderButton.tsx

```diff
@@ -1,29 +1,52 @@
 import { Button } from '@openfun/cunningham-react';
 import { useRouter } from 'next/router';
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import { Icon } from '@/components';
 import { useCreateDoc } from '@/features/docs/doc-management';
+import { useSkeletonStore } from '@/features/skeletons/store/useSkeletonStore';
 
 import { useLeftPanelStore } from '../stores';
 
 export const LeftPanelHeaderButton = () => {
   const router = useRouter();
   const { t } = useTranslation();
   const { togglePanel } = useLeftPanelStore();
+  const { setIsLoading } = useSkeletonStore();
+  const [isNavigating, setIsNavigating] = useState(false);
+
   const { mutate: createDoc, isPending: isDocCreating } = useCreateDoc({
     onSuccess: (doc) => {
-      void router.push(`/docs/${doc.id}`);
-      togglePanel();
+      setIsNavigating(true);
+      // Wait for navigation to complete
+      router.push(`/docs/${doc.id}`).then(() => {
+        // The skeleton will be disabled by the [id] page once the data is loaded
+        setIsNavigating(false);
+        togglePanel();
+      });
```

Comment:
`router.push` is a promise, it could possibly fail:
```suggestion
     router
        .push(`/docs/${doc.id}`)
        .then(() => {
          // The skeleton will be disabled by the [id] page once the data is loaded
          setIsNavigating(false);
          togglePanel();
        })
        .catch(() => {
          // In case of navigation error, disable the skeleton
          setIsLoading(false);
          setIsNavigating(false);
        });
```